WebCamSample/Node/WebCamSample/server.js

```diff
@@ -0,0 +1,335 @@
+            mediaCapture = null;
+        }
+
+        status('Initializing camera to capture audio only...');
+        mediaCapture = new Windows.Media.Capture.MediaCapture();
+        var settings = new Windows.Media.Capture.MediaCaptureInitializationSettings();
+        settings.streamingCaptureMode = Windows.Media.Capture.StreamingCaptureMode.audio;
+        settings.mediaCategory = Windows.Media.Capture.MediaCategory.other;
+        settings.audioProcessing = Windows.Media.AudioProcessing.default;
+
+        mediaCapture.initializeAsync(settings);
+
+        // Set callbacks for failure and recording limit exceeded
+        status('Device successfully initialized for audio recording!\nPress "Start Audio Record" to record');
+        mediaCapture.onfailed = mediaCapture_Failed;
+        mediaCapture.onrecordlimitationexceeded = mediaCapture_RecordLimitExceeded;
+    }
+    catch (ex) {
+        status('Unable to initialize camera for audio mode: ' + ex.Message); 
+    }
+}
+
+function takePhoto(){
+    try {
+        photoFile = Windows.Storage.KnownFolders.picturesLibrary.createFileAsync(PHOTO_FILENAME, Windows.Storage.CreationCollisionOption.generateUniqueName)
+        .then(function (storageFile) {
+            photoFile = storageFile;
+
+            var imageProperties = Windows.Media.MediaProperties.ImageEncodingProperties.createJpeg();
+            
+            mediaCapture.capturePhotoToStorageFileAsync(imageProperties, photoFile).done(function () {
+                status('Photo successfully taken and is available at ' + photoFile.path);
+            });
+        });
+    }
+    catch (ex){
+        status(ex.message);
+        cleanup();
+    }
+}
+
+function startRecordVideo(){
+    try {
+        status('Starting video recording...');
+        mode = 'video';
+
+        Windows.Storage.KnownFolders.videosLibrary.createFileAsync(VIDEO_FILENAME, Windows.Storage.CreationCollisionOption.generateUniqueName)
+        .then(function (storageFile) {
+            recordStorageFile = storageFile;
+            status('Video storage file preparation successful'); 
+            var recordProfile = Windows.Media.MediaProperties.MediaEncodingProfile.createMp4(Windows.Media.MediaProperties.VideoEncodingQuality.auto);
+
+            mediaCapture.startRecordToStorageFileAsync(recordProfile, recordStorageFile).then(function (result) {
+                isRecording = true;
+                status('Video recording in progress...press "Stop Video Record" to stop');
+            });
+        });
+    }
+    catch (ex) {
+        if (ex === Windows.System.UnauthorizedAccessException) {
+            status('Unable to play recorded video; video recorded successfully to: ' + recordStorageFile.path);
+        }
+        else {
+            status(ex.message);
+            cleanup();
+        }
+    }
+}
+
+function stopRecordVideo(){
+    try {
+        status('Stopping video recording...');
+        
+        mediaCapture.stopRecordAsync();
```

Comment:
do you need to check if recording is in progress or is it ok to stop recording anytime?